src/components/Lightbox/index.tsx

```diff
@@ -0,0 +1,222 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import type {LayoutChangeEvent, NativeSyntheticEvent, StyleProp, ViewStyle} from 'react-native';
+import {ActivityIndicator, PixelRatio, StyleSheet, View} from 'react-native';
+import Image from '@components/Image';
+import MultiGestureCanvas, {DEFAULT_ZOOM_RANGE} from '@components/MultiGestureCanvas';
+import type {CanvasSize, ContentSize, OnScaleChangedCallback, ZoomRange} from '@components/MultiGestureCanvas/types';
+import {getCanvasFitScale} from '@components/MultiGestureCanvas/utils';
+import useStyleUtils from '@hooks/useStyleUtils';
+import NUMBER_OF_CONCURRENT_LIGHTBOXES from './numberOfConcurrentLightboxes';
+
+const DEFAULT_IMAGE_SIZE = 200;
+const DEFAULT_IMAGE_DIMENSION: ContentSize = {width: DEFAULT_IMAGE_SIZE, height: DEFAULT_IMAGE_SIZE};
+
+type ImageOnLoadEvent = NativeSyntheticEvent<ContentSize>;
+
+const cachedImageDimensions = new Map<string, ContentSize | undefined>();
+
+type LightboxProps = {
+    /** Whether source url requires authentication */
+    isAuthTokenRequired?: boolean;
+
+    /** URI to full-sized attachment */
+    uri: string;
+
+    /** Triggers whenever the zoom scale changes */
+    onScaleChanged?: OnScaleChangedCallback;
```

Comment:
I am curious, why are we leaving this prop if we are not using it anymore?

Reply:
The `Lightbox` component (as well as the `MultiGestureCanvas`) are designed as generic as possible. Removing this functionality wouldn't be a good idea (imo), because we might want to use this component in other use-cases in the future. That's also why it's optional.